pkg/route/api/v1/conversion.go

```diff
@@ -13,6 +13,9 @@ func init() {
 			obj.To.Kind = "Service"
 		},
 		func(obj *TLSConfig) {
+			if len(obj.Termination) == 0 {
+				obj.Termination = TLSTerminationEdge
```

Comment:
> I don't see what validation on routes/status got to do with this.
> 
> If http will continue to function after defaulting an empty termination field to edge, why not just do it?

A route that contains a `TLSConfig` *must* have an `Termination` value or it doesn't make sense.  We want validation to enforce that so that users will be smacked when they try to create such an object.

That change makes some objects stored in etcd invalid.

Making objects in etcd invalid presents problems for controllers that try to update particular fields.  If the controller doesn't fix the "broken" field, then validation can prevent updates by the controllers which can make controllers angry.

Lucky for us, `routes/status` uses different validation, which does not require that the `Route` itself be valid.  This allows us to update the validation without changing defaulting rules and not break our controllers.